### PR TITLE
Fix internal header menu

### LINKED ISSUE #1433: Redesign homepage for the standalone site
## Problem
As Openverse will be in a standalone site, out of [wordpress.org](https://wordpress.org/), we need to update the homepage to this new context. The page needs to be more explicit about what Openverse is, how to contribute to the project, showing content about the project living inside the site, and its relation with WordPress.

Besides the above, the page needs to fix showing the content types in the content switcher by meeting a11y and i18n requirements.

Since the scope of this change is moving Openverse to a new site, current functionalities and tweaks must tend to be as minimum as possible, mainly to accomplish the primary goal described above.

## Issues related
- #991 
- #978 
- #1198 
- #1295
- #1174 
- #125 

## Proposal
https://user-images.githubusercontent.com/895819/170377079-ef058db0-bd42-4a85-bc81-96b452cb0cf2.mp4

https://user-images.githubusercontent.com/895819/170377728-d253d3f0-e859-442f-9ece-ea99e6af72fe.mp4

### Description
As you can see, compared to the current menu, this one has fewer items since I made the following changes:
- All items have no dropdown elements.
- There is a “Get involved” item linking to either the make blog or the handbook.
- Sources, Our story, Search Guide, and Meta Search pages are inside About as page’ sections instead of separated pages. I think we do not need most content from the Meta Search page as the feature should explain itself by using it, except for the paragraph describing why we have it.
- API is a page in Openverse instead of external. I suggested this time ago, but I would like your thoughts. I am not thinking of redesigning the page but placing it in the openverse site.

The main title has a light font weight to balance the page contrast and not compete with the openverse logo.

Once opening the switcher, content-type items have smaller icons to reduce the popover size, and the video option was removed. This last point is a significant change for the user experience, but the reason is not providing a content type that sets the expectation of browsing content as the others, to then finding a white page with links to external sites. The promise of searching for relevant content would not be met.

Nonetheless, the video option still will be in the search result page. I would like to see your comments on this proposal.

I displayed the sample search results in a row at the bottom and added an interaction showing the content title. I did not want to hide content depending on the viewport, but I do not have a clear idea about how it looks like on big screens.

To show that openverse relates to WordPress, I added a small note in the footer. I was thinking of adding more content but could not find any reason why do it.

On mobile, I think the best way to solve the content type labels is by hiding them and letting users tap the icon to see the options. If we show the content type names, we will have to deal with i18n all the time.

The bottom-up box to see select a content type is an idea I like, but it would work only on this page. The current flow of *leaving the site* when seeing a full-size modal to select a content type feels odd at this stage of the search journey.

This proposal has many nuances, so I am happy to hear your thoughts and try new ideas. 

## Mockup
- Figma: [3D model support and homepage redesign](https://www.figma.com/file/IFIqt7Q3MEKMLSnBedzHBO/)

### LINKED ISSUE #2105: Scroll lock and unresponsive header when navigating from home to content page on mobile
## Description

The page becomes non-interactive when you navigate from home to any content page on mobile.

## Reproduction

1. Go to https://staging.openverse.org/preferences, and make sure that `new_header` flag checkbox is on.
2. Go to https://staging.openverse.org on a mobile width
3. Click on the 'hamburger menu' in the header. In the modal, click on "About" page link.
4. See error: You won't be able to scroll down or open the header menu again. In fact, the page is non-interactive.

## Screenshots

https://user-images.githubusercontent.com/15233243/215347798-55262c40-1ea9-46e6-9531-b5fdfd25b343.mp4

## Additional context

There are two issues here, both related to insufficient cleanup of the modal on un-mounting:
- the page scroll lock should be disabled when the header un-mounts on a mobile width
- the focus trap should be deactivated when the modal is un-mounted.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1433 by @panchovm 
Fixes #2105 by @obulat

## Description

This PR adds scroll lock and focus trap cleanup on beforeUnmounting of modal in `useDialogControl`.

## Testing Instructions
Run the app. Test the app with the new header on and mobile width. You should be able to navigate from the homepage to content pages. All the links should be interactive, and all content/search/single result pages should be scrollable. Contrast that with the non-interactive content pages (eg., "About") on https://staging.openverse.org: if you navigate to About page from the homepage, the page is non-scrollable, and you cannot click any link.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: test/playwright/utils/navigation.ts [hunk 1]
```diff
@@ -464,19 +464,12 @@ export const enableNewHeader = async (page: Page) => {
...
   // Add the new_header cookie
-  await page.context().addCookies([
-    {
-      name: "features",
-      value: "%7B%22new_header%22%3A%22off%22%7D",
-      domain: "localhost",
-      path: "/",
-    },
-  ])
+  await setCookies(page.context(), { features: { new_header: "off" } })
```
Comment (dhruvkb) [inline test/playwright/utils/navigation.ts:467]: This way of setting the `new_header` flag is so much more cleaner, compared to the old approach of Playwright visiting the `/preferences/page` and toggling the checkbox!

The use of `setCookies` should be extended to the `enableNewHeader` function just above it.

**Review by dhruvkb — APPROVED** [review]
LGTM, with some non-essential suggestions!

File: src/composables/use-dialog-control.ts [hunk 2]
```diff
@@ -30,12 +35,17 @@ export function useDialogControl({
...
     triggerA11yProps["aria-expanded"] = visible
     if (shouldLockBodyScroll.value) {
       visible ? lock() : unlock()
     }
     if (!internallyControlled) emit(visible ? "open" : "close")
+    onCleanup(() => {
+      if (shouldLockBodyScroll.value) {
+        unlock()
+      }
+    })
```
Comment (zackkrida) [inline src/composables/use-dialog-control.ts:48]: Very nice

File: package.json [hunk 3]
```diff
@@ -43,7 +43,7 @@
     "test:playwright:debug": "PWDEBUG=1 pnpm test:playwright:local",
     "test:playwright:recreate-tapes": "rimraf test/tapes && pnpm test:playwright:update-tapes",
     "test:playwright:update-tapes": "UPDATE_TAPES=true pnpm test:playwright",
-    "test:playwright:gen": "playwright codegen localhost:8443/",
+    "test:playwright:gen": "playwright codegen --viewport-size=340,600 localhost:8443/",
```
[the remark this answers is not in the document]
Comment (obulat) [reply package.json]: Moved to the Playwright readme.

**Review by zackkrida — COMMENTED** [review]
LGTM!

**Review by zackkrida — APPROVED** [review]
LGTM!

**Pull request merged**